runtime/server/x86/decoder/ctc_prefix_beam_search.cc

```diff
@@ -0,0 +1,128 @@
+// Copyright 2020 Mobvoi Inc. All Rights Reserved.
+// Author: binbinzhang@mobvoi.com (Binbin Zhang)
+
+#include <glog/logging.h>
+
+#include <algorithm>
+#include <tuple>
+#include <unordered_map>
+#include <utility>
+#include <cmath>
+
+#include "decoder/ctc_prefix_beam_search.h"
+
+namespace wenet {
+
+CtcPrefixBeamSearch::CtcPrefixBeamSearch(
+    const CtcPrefixBeamSearchOptions& opts): opts_(opts) {
+  Reset();
+}
+
+void CtcPrefixBeamSearch::Reset() {
+  hypotheses_.clear();
+  likelihood_.clear();
+  cur_hyps_.clear();
+  PrefixScore prefix_score(0.0, -std::numeric_limits<float>::max());
+  std::vector<int> empty;
+  cur_hyps_[empty] = prefix_score;
+}
+
+static float LogAdd(float x, float y) {
+  const float kMinLogDiffFloat = std::log(1.19209290e-7f);
```

Comment:
std::logf

Reply:
there is no logf in std since it is a overrided function.

Reply:
Of course, there is `std::logf`!

Please google it.
https://en.cppreference.com/w/cpp/numeric/math/log